kernel/src/actions/deletion_vector.rs

```diff
@@ -223,15 +229,111 @@ pub(crate) fn treemap_to_bools(treemap: RoaringTreemap) -> Vec<bool> {
     }
 }
 
+#[derive(PartialEq)]
+#[repr(C)]
+pub enum VectorType {
+    Selection,
+    Deletion,
+}
+
+pub trait BitSet<V> {
+    fn set_bit(&mut self, idx: usize, value: impl Into<V>);
+}
+
+pub trait ContainerIterator<C, V>
```

Comment:
Do we expect to have other things that implement this? If so, what? If not, it feels like just unneeded extra abstraction. We can just have these methods directly on `DeletionVectorBatch`